discovery/gossiper.go

```diff
@@ -555,6 +553,10 @@ type AuthenticatedGossiper struct {
 	vb *ValidationBarrier
 
 	sync.Mutex
+
+	cancel fn.Option[context.CancelFunc]
```

Comment:
Q: I don't think we want to allow this to be optional?

Reply:
the reason we need to make it optional is cause it is something we set at `Start` time when we derive it from a parent context. Only things that are set in the constructor can we for sure know are set at `Stop` time. It needs to be safe to call `Stop` even if `Start` has not run.

Reply:
> It needs to be safe to call Stop even if Start has not run.

Why is that? If it's not started why do we even call `Stop`?

Reply:
because of the changes made in this PR from last year: https://github.com/lightningnetwork/lnd/commit/653e2f36670477a64351cbe3a4aa223e6ab50107